Add recipe for Classy Vision

mannatsingh opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in 
a comment on the PR to get the attention of the `staged-recipes` team. You can 
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/classy_vision```) and found it was in an excellent condition.

mannatsingh commented:
I am willing to be listed as a maintainer

vreis commented:
I am willing to be listed as a maintainer

mannatsingh commented:
This PR is ready for review.

@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

chrisburr commented on recipes/classy_vision/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+
+build:
+  number: 0
+  skip: True  # [py<36 or not linux64]
```
Can this be noarch? It's okay for non linux builds to fail
```suggestion
  noarch: python
```

chrisburr commented on recipes/classy_vision/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+requirements:
+  build:
+  host:
+    - python
```
If (and only if!) it is made noarch:
```suggestion
    - python >=3.6
```

chrisburr commented on recipes/classy_vision/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+    - python
+    - pip
+  run:
+    - python
```
If (and only if!) it is made noarch:
```suggestion
    - python >=3.6
```

mannatsingh commented:
@chrisburr Thanks for the quick review! I hadn't used noarch python since one of the constraints mentioned in https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python wasn't satisfied ("Scripts argument in setup.py is not used" - see https://github.com/facebookresearch/ClassyVision/blob/master/setup.py#L64). But I converted to noarch and the linux build still works. I added a test to make sure the script works, and it does. 

We can continue with noarch if you think it's fine, otherwise I can go back to the previous state.

chrisburr commented:
> @chrisburr Thanks for the quick review! I hadn't used noarch python since one of the constraints mentioned in https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python wasn't satisfied ("Scripts argument in setup.py is not used" - see https://github.com/facebookresearch/ClassyVision/blob/master/setup.py#L64). But I converted to noarch and the linux build still works. I added a test to make sure the script works, and it does.
> 
> We can continue with noarch if you think it's fine, otherwise I can go back to the previous state.

You're right, it can't be noarch unfortunately. If you revert 804a831 I'll merge.
